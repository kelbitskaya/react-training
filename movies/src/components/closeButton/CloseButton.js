import React from 'react';
import PropTypes from 'prop-types';

export default function CloseButton(props) {
  const { handleClose } = props;
  return (
      <button
        type="button"
        className="button-close"
        onClick={handleClose}/>
  );

}
CloseButton.propTypes = {
  handleClose: PropTypes.func.isRequired,
};
