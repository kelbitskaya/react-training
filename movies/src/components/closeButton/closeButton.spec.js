// import React from 'react';
// import CloseButton from './CloseButton';
//
// const props = {
//   handleClose: ()=>{},
// };
//
// const setUp = (props) => shallow(<CloseButton {...props}/>);
//
// describe("CloseButton component", () => {
//   describe("has props", () => {
//     const component = setUp(props);
//     it("should render CloseButton component", ()=> {
//       const button = component.find("button");
//       expect(button).toHaveLength(1);
//     });
//   });
// });

import React from 'react';
import rerender from 'react-test-renderer';
import CloseButton from './CloseButton';

describe('CloseButton component', () => {
  let handleClose;
  beforeEach(()=>{
    handleClose = jest.fn();
  });
  test('CloseButton layout', () => {
    const component = rerender.create(<CloseButton handleClose={handleClose}/>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('CloseButton', () => {
    const closeButton = shallow(<CloseButton  handleClose={handleClose}/>);
    closeButton.find('.button-close').simulate('click');
    closeButton.simulate('click');
    expect(handleClose).toBeCalled();
  });

});
