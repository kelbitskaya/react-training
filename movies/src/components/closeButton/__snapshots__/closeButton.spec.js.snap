// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton component CloseButton layout 1`] = `
<button
  aria-label="Close"
  className="button-close"
  onClick={[MockFunction]}
  type="button"
/>
`;
